### PR TITLE
config: add included files into parsedContents before applying

### LINKED ISSUE #8719: `include` directive in nix.conf has no effect
**Describe the bug**

The `include` directive in nix.conf has no effect in 2.16.1 (also tested in current master 453c4be93cb10b417a7140d9f8b7185f9785b7a6, same result)

**Steps To Reproduce**

```
# /etc/nix/nix.conf
substituters = https://cache.nixos.org/
include nix.local.conf
```
```
# /etc/nix/nix.local.conf
extra-substituters = http://foo.example
```

**Expected behavior**

Results of `nix show-config`:

- Expected to include: `substituters = https://cache.nixos.org/ http://foo.example`
- Actually includes: `substituters = https://cache.nixos.org/`

**`nix-env --version` output**

nix-env (Nix) 2.16.1

**Additional context**

The issue does not occur on Nix 2.15.1. There is no `{~,~root}/.config/nix/nix.conf`.

**Priorities**

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

### PR DESCRIPTION
**Pull request by bouk**
Fixes #8719

### CODE REVIEW

File: src/libutil/config.cc [hunk 1]
```diff
@@ -123,7 +122,10 @@ void AbstractConfig::applyConfig(const std::string & contents, const std::string
                 throw UsageError("illegal configuration line '%1%' in '%2%'", line, path);
             auto p = absPath(tokens[1], dirOf(path));
             if (pathExists(p)) {
-                applyConfigFile(p);
+                try {
+                    std::string includedContents = readFile(path);
+                    applyConfigInner(includedContents, p, parsedContents);
+                } catch (SysError &) { }
```
Comment (bouk) [inline src/libutil/config.cc:128]: I added the try/catch so it matches `applyConfigFile`, not sure it's needed

Comment (Ericson2314) [reply src/libutil/config.cc:128]: Yeah good call. It does technically make a difference if the file has funny perms.

Write a TODO for "do we actually want this? or is it better to fail?" maybe? I think it was an accident. But best actually removed in a separate PR.

**Comment by Ericson2314** [comment]
Now `applyConfigFile` is just used twice in one function: `applyConfigFile` in `src/libstore/globals.cc`. Can you delete `applyConfigFile` and just make it a lambda inside `applyConfigFile` instead?

**Review by Ericson2314 — CHANGES_REQUESTED** [review]
Great so far! Just those two more things.

**Comment by bouk** [comment]
@Ericson2314 addressed your comments

**Review by Ericson2314 — APPROVED** [review]
Looks great!

**Pull request merged**